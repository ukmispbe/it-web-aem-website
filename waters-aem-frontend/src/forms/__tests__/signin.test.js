import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Form from '../form';

const config =
    {
        "submitEndpoint": "https://test-www.waters.com:8443/api/waters/user/v1/login",
        "siteKey": "6Ld5WMIUAAAAACZQvEc7I75aEg5AC8YUUO0W7zRG",
        "formName": "signin",
        "buttonText": "SIGN IN",
        "icons": {
            "checkmarkIcon": "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
            "validIcon": "/content/dam/waters/en/brand-assets/icons/checkmark.svg",
            "invalidIcon": "/content/dam/waters/en/brand-assets/icons/attention.svg",
            "eyeIcon": "/content/dam/waters/en/brand-assets/icons/eye.svg",
            "eyeOffIcon": "/content/dam/waters/en/brand-assets/icons/eye-off.svg",
            "signInIcon": "/content/dam/waters/en/brand-assets/icons/user.svg"
        },
        "fields": [
            {
                "type": "email",
                "name": "email",
                "label": "Email Address",
                "validation": {
                    "required": true,
                    "validateFnName": "email",
                    "requiredMsg": "Please enter a valid email address.",
                    "validationMsg": "Please enter a valid email address."
                }
            },
            {
                "type": "password",
                "name": "password",
                "label": "Password",
                "validation": {
                    "required": true,
                    "validateFnName": "noValidation",
                    "requiredMsg": ""
                }
            },
            {
                "type": "link",
                "name": "forgotPassword",
                "text": "Having trouble signing in?",
                "link": "/nextgen/be.html"
            }
        ]
    };

const submitFn = jest.fn();
const isocode = 'en_us';

describe('Feature: Sign In Form', () => {
    describe('Scenario: Rendering', () => {
        describe('When initial render', ()=>{
            it('Then it should render an email address field', async ()=>{
                let wrapper;
                await act(async () => {
                    wrapper = mount(<Form config={config} submitFn={submitFn} isocode={isocode} />);
                });

                const label = wrapper.find('label[htmlFor="email"]');
                expect(label.exists()).toEqual(true);

                const input = wrapper.find('input[name="email"]');
                expect(input.exists()).toEqual(true);
                wrapper.unmount();
            })
            it('Then it should render a password field', async ()=>{
                let wrapper;
                await act(async () => {
                    wrapper = mount(<Form config={config} submitFn={submitFn} isocode={isocode} />);
                });

                const label = wrapper.find('label[htmlFor="password"]');
                expect(label.exists()).toEqual(true);

                const input = wrapper.find('input[name="password"]');
                expect(input.exists()).toEqual(true);
                wrapper.unmount();
            })
            it('Then it should render a link', async ()=>{
                let wrapper;
                await act(async () => {
                    wrapper = mount(<Form config={config} submitFn={submitFn} isocode={isocode} />);
                });

                const link = wrapper.find('div.cmp-form-field-link--forgotPassword').find('a');
                expect(link.exists()).toEqual(true);
                wrapper.unmount();
            })
            it('Then it should render a submit button', async ()=>{
                let wrapper;
                await act(async () => {
                    wrapper = mount(<Form config={config} submitFn={submitFn} isocode={isocode} />);
                });

                const button = wrapper.find('button');
                expect(button.exists()).toEqual(true);
                wrapper.unmount();
            })
        })
    })
})