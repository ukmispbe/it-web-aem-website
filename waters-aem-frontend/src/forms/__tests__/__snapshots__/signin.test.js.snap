// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: Sign In Form Scenario: Rendering When initial render Then the snapshot should match 1`] = `
<form
  className="cmp-form cmp-form--registration "
  data-locator="form-component"
  onSubmit={[Function]}
>
  <div
    className="cmp-form-field cmp-form-field-email"
  >
    <label
      className=""
      data-locator="email-address"
      htmlFor="email"
    >
      Email Address
      <span
        className="cmp-form-field--required"
      >
        *
      </span>
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        aria-label="email"
        aria-labelledby="email"
        aria-required={true}
        className=""
        data-locator="email"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="email"
      />
      <div
        className="cmp-form-field--icons"
      >
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-password"
  >
    <label
      className=""
      data-locator="password"
      htmlFor="password"
    >
      Password
      <span
        className="cmp-form-field--required"
      >
        *
      </span>
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        aria-label="password"
        aria-labelledby="password"
        aria-required={true}
        className=""
        data-locator="password"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="password"
      />
      <div
        className="cmp-form-field--icons"
      >
        <div
          className="showHide-icon toggled"
          onMouseDown={[Function]}
        />
        <div
          className="showHideOff-icon"
          onMouseDown={[Function]}
        />
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-link"
  >
    <div
      className="cmp-form-field-link--forgotPassword "
    >
      <a
        data-locator="having-trouble-signing-in?"
        href="/nextgen/be.html"
        rel="noopener noreferrer"
        target=""
      >
        Having trouble signing in?
      </a>
    </div>
  </div>
  <button
    className="cmp-button cmp-button--no-border cmp-form--submit"
    data-locator="form-submit"
    type="submit"
  >
    SIGN IN
  </button>
</form>
`;
